### PR TITLE
Wrong error message in logs

### LINKED ISSUE #9866: Error "Frontcontroller::init - Cart cannot be loaded or an order has already been placed using this cart" in logs while using the reorder feature
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-4009).

- _**Reporter:**_ salma.moakhar
- _**Created at:**_ Thu, 12 Oct 2017 15:22:39 +0200

`Frontcontroller::init - Cart cannot be loaded or an order has already been placed using this cart`

This happens whenever an order is placed using any payment module. However, everything is working fine, orders are getting placed and nothing is missing.

I thought that this whole ordeal was fixed since the creation of this topic:
https://www.prestashop.com/forums/topic/466782-cart-cannot-be-loaded-or-an-order-has-already-been-placed-using-this-cart/

..but I guess I was wrong. Just thought I'd raise awareness so the devs can get to fixing this once and for all.
        
---
#### Describe the bug

This happens whenever an order is placed using any payment module. However, everything is working fine, orders are getting placed and nothing is missing.

#### Expected behavior

No error displayed in the logs

#### Steps to Reproduce

1. Go to **FO**
2. Create an **Order**
3. Choose any **payment method** and validate your order
4. Got to **BO** > **CONFIGURE** > **Advanced Parameters** > **Logs**
5. See error

**Screenshots**

image

#### Additional information

* PrestaShop version: 1.7.4.2 -> 1.7.7.3
* PHP version: 7.2

### LINKED ISSUE #27838: Wrong error message in logs
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

The problem is the following error message:
`Frontcontroller::init - Cart cannot be loaded or an order has already been placed using this cart`

The PHP test performed to display this message does not match this one.

### Expected behavior

Change the display condition of the message to better match the text

### Steps to reproduce

You will find below the steps to display the error messages in the backoffice.

1. Have 2 tabs open on the "shopping cart ?action=show" page (with items)
image

1. Launch and complete the order on one of the tabs
image

1. Refresh the other tab still active on the old cart 

1. Go to the back-office in "Advanced settings --> Logs" and the following error message should appear 
image

### PrestaShop version(s) where the bug happened

All

### PHP version(s) where the bug happened

All

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | The problem is the following error message:<br/>`Frontcontroller::init - Cart cannot be loaded or an order hasalready been placed using this cart`<br/>The PHP test performed to display this message does not match this one.
| Type?             |  bug fix
| Category?         | BO
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | Fixes #9866
| How to test?      | See the issues `Steps to reproduce`
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27839)

### CODE REVIEW

**Review by atomiix — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by PululuK — DISMISSED** [review]

**Review by HanaRebaiQA — DISMISSED** [review]
Hello @tlgsoledis 

I have tested this PR. The issue is not fixed using these steps : 

1. Go to FO
2. Create an Order
3. Choose any payment method and validate the order
4. Got to BO > CONFIGURE > Advanced Parameters > Logs
See error

image

https://user-images.githubusercontent.com/92912932/157213326-d6c28e7b-6a1f-45f1-848e-173b1e39afa9.mp4


I also detect this issue using the same steps : 
image

All caches from browser and from PS in BO > CONFIGURE > Advanced Parameters > Performance are cleared.
Even, i tried with a private navigation.

Could you please check and feedback ?

Thanks!

**Comment by ghost** [comment]
Hello @HanaRebaiQA,
I have done the same test locally and indeed, I have the same problem.
I made a new modification in my commit 880dcb0bf2d66ed647ef3f112f59e04abd78fadc which corrects the appearance of this log in this context.

File: controllers/front/OrderConfirmationController.php [hunk 1]
```diff
@@ -49,6 +49,15 @@ class OrderConfirmationControllerCore extends FrontController
      */
     public function init()
     {
+        //Test below to prevent unnecessary logs from "parent::init()"
+        $this->id_cart = (int)(Tools::getValue('id_cart', 0));
```
Comment (Progi1984) [inline controllers/front/OrderConfirmationController.php]:
```suggestion
        $this->id_cart = (int) Tools::getValue('id_cart', 0);
```

File: controllers/front/OrderConfirmationController.php [hunk 2]
```diff
@@ -49,6 +49,15 @@ class OrderConfirmationControllerCore extends FrontController
      */
     public function init()
     {
+        //Test below to prevent unnecessary logs from "parent::init()"
```
Comment (Progi1984) [inline controllers/front/OrderConfirmationController.php]:
```suggestion
        // Test below to prevent unnecessary logs from "parent::init()"
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks but it's good.

File: controllers/front/OrderConfirmationController.php (hunk 1 above)
Comment (kpodemski) [reply controllers/front/OrderConfirmationController.php]: @tlgsoledis you forgot about this one

**Comment by ghost** [comment]
@Progi1984 i have modified the files according to the requested standards.

**Comment by ghost** [comment]
Hello @Progi1984 @PululuK @atomiix 
Can you please restart the code review.
Thanks in advance.

**Review by atomiix — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by HanaRebaiQA — COMMENTED** [review]
Hello @tlgsoledis 

I checked again this PR.
The same issues previously mentioned still exist.

-There are 3 message errors
-A new error message is also dispalyed 

I tested with multistore and same problem.

image

https://user-images.githubusercontent.com/92912932/161031627-5eef148e-c987-4d74-8d42-8f88b96383bc.mp4



==> The expected is that the Severity still `Informative only `and the Object should be `Cart`  but without the error message `Frontcontroller::init - Cart cannot be loaded or an order hasalready been placed using this cart`

Could you please check ?

Thanks!

**Comment by ghost** [comment]
Hello @HanaRebaiQA 
The logs raised in your test have nothing to do with the fix I made.
They probably come from a base module that always uses the "Tools::redirectLink()" method.
For the errors on "Swift Error", it probably comes from the fact that you did not set up the email sending correctly in your shop.

**Comment by atomiix** [comment]
@tlgsoledis can you remove your last commit? It seems that it's an empty commit, that is a **merge** commit and we don't allow merge commit in PRs 🙂

**Comment by ghost** [comment]
Hello @atomiix,
I do the revert commit for my merge request.
Sorry for this.

**Comment by kpodemski** [comment]
Hello @tlgsoledis 

Something went wrong with your rebase.

**Comment by ghost** [comment]
Hello @atomiix @kiropowered ,
I think I have done the necessary backtracking.

**Review by kpodemski — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by HanaRebaiQA — APPROVED** [review]
Hello @tlgsoledis 

I checked again this PR. The issue is fixed.
Here my tested cases : 

- Check the order confirmation in BO and FO : OK
- Check with different payment methods : OK
- Check with different languages (RTL and no RTL) : OK
- Check with multistore : OK

https://user-images.githubusercontent.com/92912932/161967999-6e62def1-287f-4dc8-b4f5-2f13c20044e5.mp4


So, it's QA :heavy_check_mark: 

Thanks!

**Comment by Progi1984** [comment]
Thanks @tlgsoledis & @HanaRebaiQA

**Review by awitkutarahil — APPROVED** [review]

**Comment by ghtoo** [comment]
It seems that some payment modules initialize FrontController, and while there is order already placed and cart cookie present, Prestashop adds this log message. This particular fix only makes ps bank wire module not to produce aforementioned warning.

**Comment by ghost** [comment]
Hello @ghtoo 
Can you provide me with an example of a payment module that would still issue the log with the correction made?

**Pull request merged**